ext/date/php_date.c

```diff
@@ -876,17 +876,16 @@ PHPAPI int php_idate(char format, time_t ts, bool localtime)
 		}
 	}
 
-	timelib_isoweek_from_date(t->y, t->m, t->d, &isoweek, &isoyear);
-
 	switch (format) {
 		/* day */
 		case 'd': case 'j': retval = (int) t->d; break;
 
+		case 'N': retval = (int) timelib_iso_day_of_week(t->y, t->m, t->d); break;
 		case 'w': retval = (int) timelib_day_of_week(t->y, t->m, t->d); break;
 		case 'z': retval = (int) timelib_day_of_year(t->y, t->m, t->d); break;
 
 		/* week */
-		case 'W': retval = (int) isoweek; break; /* iso weeknr */
+		case 'W': timelib_isoweek_from_date(t->y, t->m, t->d, &isoweek, &isoyear); retval = (int) isoweek; break; /* iso weeknr */
```

Comment:
I don't approve of multiple statements on this line here. You'd have to break into multiple... and hence, just leave the call where it was in line 879.

Reply:
Better, yes, but the comment should be moved right behind the case label. OTOH, if @derickr, who is the maintainer of the date extension, prefers to call that function unconditionally (I assume it is a rather cheap function call), than it should be done that way. :)

Reply:
No, please put it back like it was.

Reply:
You mean TIMELIB_UNSET, or the timelib_isoweek_from_date call, or both?

If you mean the timelib_isoweek_from_date call, mind explaining why it's better to always call it even when unneeded (which date() function doesn't do)?

Reply:
@derickr Can you explain how you want to proceed?

I can do either change, I just want to understand what the reason is.

Reply:
@derickr prefers calling `timelib_isoweek_from_date()` unconditionally, like it was before.